airflow/cli/commands/dag_processor_command.py

```diff
@@ -19,30 +19,40 @@
 
 import logging
 from datetime import timedelta
+from typing import Any
 
 import daemon
 from daemon.pidfile import TimeoutPIDLockFile
 
 from airflow import settings
 from airflow.configuration import conf
-from airflow.jobs.dag_processor_job import DagProcessorJob
+from airflow.jobs.base_job import BaseJob
+from airflow.jobs.dag_processor_job import DagProcessorJobRunner
 from airflow.utils import cli as cli_utils
 from airflow.utils.cli import setup_locations, setup_logging
 
 log = logging.getLogger(__name__)
 
+dag_processor_job_runner: DagProcessorJobRunner | None = None
 
-def _create_dag_processor_job(args) -> DagProcessorJob:
+
+def _create_dag_processor_job(args: Any) -> BaseJob:
     """Creates DagFileProcessorProcess instance."""
+    from airflow.dag_processing.manager import DagFileProcessorManager
+
     processor_timeout_seconds: int = conf.getint("core", "dag_file_processor_timeout")
     processor_timeout = timedelta(seconds=processor_timeout_seconds)
-    return DagProcessorJob(
+    global dag_processor_job_runner
+    processor = DagFileProcessorManager(
         processor_timeout=processor_timeout,
         dag_directory=args.subdir,
         max_runs=args.num_runs,
         dag_ids=[],
         pickle_dags=args.do_pickle,
     )
+    return BaseJob(
+        job_runner=processor.job_runner,
+    )
```

Comment:
Code here looks wrong? `dag_processor_job_runner` is never assigned anywhere.

FWIW I also dislike the global mutable variable design. A `get_dag_processor` function with `lru_cache` would be significantly better.

Reply:
I actually removed it later in the following PRs it was only needed temporaryily to fix some of the tests (final version does not need it). I will remove it from here as well - I think I can do it now after some other changes - that could have been a remnant that is not really needed.

Reply:
Just to explain - with this kind of staged refactor, some things were added just to make each of the steps smaller and they are really temporary - this is the case here. I could have also merged it all in one big PR, but this one big PR would be rather difficult to review in one go (and when I did it in the past I was asked to split it into reviewable pieces). But I can see if I can remove that one now looking at the final stage without addign too much to this PR.

Reply:
Just a thought @uranusjr -> but maybe indeed it is a good idea to just review 
https://github.com/apache/airflow/pull/30376 ? 

This is a merge of those 4 PRS which I tried to keep separate and address one part of the refactor at a time. But the final stage is "WAY" cleaner than any of the intermediate steps. Those intermediate steps were only really needed to make sure that there is a "surgical" change at each of the steps while the ORM -> logic relation had been unentangled and to keep the tests green (and to test each step separately)

WDYT?

Reply:
OK. I got rid of the global state in this PR and rebasing others on top anyway - regardless of the way we continue with the review.

Reply:
(FYI. Yes. The global variable has not been set - I "fixed" that in the following PR #30302 actually (I found that during resolving conflicts). But since that has not been "really" used without AIP-44 applied (Pydantic path is not yet used) and I got it removed in #30376 anyway, so removing the whole part there could be removed.